src/main/webapp/app/entities/feedback.model.ts

```diff
@@ -53,13 +53,17 @@ export class Feedback implements BaseEntity {
     }
 
     public static isEmpty(that: Feedback): boolean {
-        return that.credits === 0 && !Feedback.hasDetailText(that);
+        return that.credits == undefined || !Feedback.hasDetailText(that);
     }
 
-    public static isPresent(that: Feedback): boolean {
+    public static isValid(that: Feedback): boolean {
         return !Feedback.isEmpty(that);
     }
 
+    public static areValid(that: Feedback[]): boolean {
+        return that.filter(Feedback.isValid).length > 0 && that.filter(Feedback.isValid).length === that.length;
```

Comment:
Since this only evaluates to true if the lengths match, this can simplify to `return that.length > 0 && that.filter(Feedback.isValid).length === that.length;` which saves a tiny bit of performance by not filtering twice.

Reply:
The check  `that.filter(Feedback.isValid).length === that.length` is needed to verify that ALL feedback items are valid. With `.length > 0` it validets true when at least one feedback element is valid. :)